src/adapter/src/coord/peek.rs

```diff
@@ -287,7 +287,10 @@ fn permute_oneshot_mfp_around_index(
     let input_arity = mfp.input_arity;
     let mut safe_mfp = mfp_to_safe_plan(mfp)?;
     let (permute, thinning) = mz_expr::permutation_for_arrangement(key, input_arity);
-    safe_mfp.permute(permute, key.len() + thinning.len());
+    safe_mfp.permute(
+        permute.into_iter().enumerate().collect(),
+        key.len() + thinning.len(),
+    );
```

Comment:
Can we change `permute` to accept a `&[usize]` instead? If I'm reading this correctly, all uses are of the form `permute(permute.into_iter().enumerate().collect(), ...)`, unless there are some outside of this change that I can't see here!

Reply:
Unfortunately it's a bit more complicated. These call in to `MapFilterProject::permute`, which leads with 
```rust
        for index in 0..map.len() {
            // Intermediate columns are just shifted.
            shuffle.insert(self.input_arity + index, new_input_arity + index);
        }
```
so .. as currently written .. it wants mutable access to a map in order to store remappings that the caller is unaware of. We can potentially make it closure based, though, and I'll take a look at that.